src/services/refactors/convertImport.ts

```diff
@@ -0,0 +1,113 @@
+/* @internal */
+namespace ts.refactor.generateGetAccessorAndSetAccessor {
+    const refactorName = "Convert import";
+    const actionNameNamespaceToNamed = "Convert namespace import to named imports";
+    const actionNameNamedToNamespace = "Convert named imports to namespace import";
+    registerRefactor(refactorName, {
+        getAvailableActions(context): ApplicableRefactorInfo[] | undefined {
+            const i = getImportToConvert(context);
+            if (!i) return undefined;
+            const description = i.kind === SyntaxKind.NamespaceImport ? Diagnostics.Convert_namespace_import_to_named_imports.message : Diagnostics.Convert_named_imports_to_namespace_import.message;
+            const actionName = i.kind === SyntaxKind.NamespaceImport ? actionNameNamespaceToNamed : actionNameNamedToNamespace;
+            return [{ name: refactorName, description, actions: [{ name: actionName, description }] }];
+        },
+        getEditsForAction(context, actionName): RefactorEditInfo {
+            Debug.assert(actionName === actionNameNamespaceToNamed || actionName === actionNameNamedToNamespace);
+            const edits = textChanges.ChangeTracker.with(context, t => doChange(context.file, context.program.getTypeChecker(), t, Debug.assertDefined(getImportToConvert(context))));
+            return { edits, renameFilename: undefined, renameLocation: undefined };
+        }
+    });
+
+    // Can convert imports of the form `import * as m from "m";` or `import d, { x, y } from "m";`.
+    function getImportToConvert(context: RefactorContext): NamedImportBindings | undefined {
+        const { file } = context;
+        const span = getRefactorContextSpan(context);
+        const token = getTokenAtPosition(file, span.start, /*includeJsDocComment*/ false);
+        const importDecl = getParentNodeInSpan(token, file, span);
+        if (!importDecl || !isImportDeclaration(importDecl)) return undefined;
+        const { importClause } = importDecl;
+        return importClause && importClause.namedBindings;

```

Comment:
should we verify that the namedBindings is within the span..

e.g. selecting `d` in `import d, * as ns from "./mod"` should not trigger any action..